docs/spec/supervisor/supervisor.md

```diff
@@ -0,0 +1,153 @@
+
+```rust
+/// Stores the Light Blocks which could not be confirmed to have originated from
+/// the same chain as the predecesor light blocks.
+type FetchedStack = Stack<LightBlock>;
+
+/// Pair of light block and height of the light block which was highest 
+/// highest one for which we believe originated from the same chain
+type ChainPair = (LightBlock, Height);
+
+/// Traces the history of verify_to_target.
+trait Trace {
+    /// Returns the height of the current highest inserted
+    fn current_height(&self) -> Height;
+
+    /// Returns the current highest inserted light block.
+    fn current(&self) -> &LightBlock;
+
+    /// Insert a new highest light block. assert(self.current_height() < light_block.height())
+    fn insert(&mut self, light_block: LightBlock) -> Result<(), Error>;
+
+    /// Returns the list of all LightBlock which were inserted through 
+    /// insert with associated previous current_height().
+    fn verification_chain(&self) -> LinkedList<ChainPair>;
+}
+
+
+/// Given a primary and a starting_light_block confirm that the light block of target_height received
+/// from primary for target_height is from the same chain as the starting_light_block.
+/// Solution for forward skipping verification.
+pub fn verify_to_target(
+    primary: PeerID,
+    starting_light_block: LightBlock,
+    target_height: Height) -> Result<Trace, (LightBlock, Reason)> {
```

Comment:
The `Reason` type is not expanded anywhere. Would be helpful to see what it actually looks like. It's uncommon to not

Reply:
It's a mistake it should be [`Error`](https://github.com/informalsystems/tendermint-rs/blob/548d76d8b7025f1a3ad1ee9adf32a5e6d24e2197/light-client/src/errors.rs#L18).